internal/resources/scope.go

```diff
@@ -0,0 +1,347 @@
+		ProjectId:      types.StringValue(projectId),
+		ClusterId:      types.StringValue(clusterId),
+		BucketId:       types.StringValue(bucketId),
+	}
+
+	objectList := make([]types.Object, 0)
+	//traverse collections
+	for _, apiCollection := range *scopeResp.Collections {
+		//create object
+		providerschemaCollection := providerschema.NewCollection(apiCollection)
+		collectionObj, diag := types.ObjectValueFrom(ctx, providerschema.CollectionAttributeTypes(), providerschemaCollection)
+		if diag.HasError() {
+			return nil, fmt.Errorf("collection object error"), diag
+		}
+		//print data to check apiCollection
+		data, _ := json.Marshal(apiCollection)
+		fmt.Println(string(data))
+
+		objectList = append(objectList, collectionObj)
+	}
+
+	//create collection set
+	//using a set instead of list as we need unique list of collections, no duplicates
+	collectionSet, diag := types.SetValueFrom(ctx, types.ObjectType{}.WithAttributeTypes(providerschema.CollectionAttributeTypes()), objectList)
+	if diag.HasError() {
+		return nil, fmt.Errorf("collection set error"), diag
+	}
+
+	refreshedState.Collections = collectionSet
+
+	return &refreshedState, nil, nil
+}
+
+func validateScopeNameIsSameInPlanAndState(planScopeName, stateScopeName string) bool {
+	return strings.EqualFold(planScopeName, stateScopeName)
+}
+
+func (s *Scope) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.Scope
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Scope in Capella",
+			"Could not read Capella Scope with name "+state.Name.String()+": "+err.Error(),
+		)
+		return
+	}
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		bucketId       = IDs[providerschema.BucketId]
+		scopeName      = IDs[providerschema.ScopeName]
+	)
+
+	refreshedState, err, diag := s.retrieveScope(ctx, organizationId, projectId, clusterId, bucketId, scopeName)
+	if diag.HasError() {
+
+		diags.Append(diag...)
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error reading scope",
+			"Could not read scope name "+state.Name.String()+": "+errString,
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, &refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (s Scope) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	//TODO implement me
+	panic("implement me")
```

Comment:
If there is no update supported, we can leave this function empty. We can refer to allowedCidrs resource for guidance.